pkg/registry/servicecatalog/broker/storage.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/registry/generic"
+	"k8s.io/kubernetes/pkg/registry/generic/registry"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+// Match determines whether an Instance matches a field and label
+// selector.
+func Match(label labels.Selector, field fields.Selector) storage.SelectionPredicate {
+	return storage.SelectionPredicate{
+		Label:    label,
+		Field:    field,
+		GetAttrs: GetAttrs,
+	}
+}
+
+// ToSelectableFields returns a field set that represents the object for matching purposes.
+func ToSelectableFields(broker *servicecatalog.Broker) fields.Set {
+	objectMetaFieldsSet := generic.ObjectMetaFieldsSet(&broker.ObjectMeta, true)
+	return generic.MergeFieldsSets(objectMetaFieldsSet, nil)
+}
+
+// GetAttrs returns labels and fields of a given object for filtering purposes.
+func GetAttrs(obj runtime.Object) (labels.Set, fields.Set, error) {
+	broker, ok := obj.(*servicecatalog.Broker)
+	if !ok {
+		return nil, nil, fmt.Errorf("given object is not a Broker")
+	}
+	return labels.Set(broker.ObjectMeta.Labels), ToSelectableFields(broker), nil
+}
+
+// NewStorage creates a new rest.Storage responsible for accessing Instance
+// resources
+func NewStorage(opts generic.RESTOptions) rest.Storage {
+	prefix := "/" + opts.ResourcePrefix
+
+	store := registry.Store{
+		NewFunc: func() runtime.Object {
+			return &servicecatalog.Broker{}
+		},
+		// NewListFunc returns an object capable of storing results of an etcd list.
+		NewListFunc: func() runtime.Object {
+			return &servicecatalog.BrokerList{}
+		},
+		// Produces a path that etcd understands, to the root of the resource
+		// by combining the namespace in the context with the given prefix
+		KeyRootFunc: func(ctx api.Context) string {
```

Comment:
Generic design note. We don't seem to be doing anything besides calling a builtin with a small customization here(`prefix`). If k8s is supposed to be all about convention over configuration, this stuff seems to be all about the opposite. Configure all the things, even if the default would get us most of the way there.  I'd think the store should take a `prefix` and then `Key*Func`s go away.

I don't know enough go, but I would hope there was a way to give the store a `Broker` and it creates the New, NewList, ObjectName, and maybe even Qualified resource. My limited experience makes this look very boilerplate. Future codegen opportunity?